aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -241,11 +251,14 @@ private Completable transitionToApiSync(Mode current) {
     private void startObservingStorageChanges() {
         LOG.info("Starting to observe local storage changes.");
         try {
-            mutationOutbox.load()
+            boolean subscribed = mutationOutbox.load()
                 .andThen(Completable.create(emitter -> {
                     storageObserver.startObservingStorageChanges(emitter::onComplete);
                     currentMode.set(Mode.LOCAL_ONLY);
-                })).blockingAwait(OP_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+                })).blockingAwait(adjustedTimeoutSeconds, TimeUnit.SECONDS);
+            if (!subscribed) {
+                throw new TimeoutException("Subscription timed out.");
+            }
```

Comment:
I could be wrong, here, but I _believe_ this timeout should be _way_ shorter. Starting the subscription to the local storage should be O(ms), to complete a CPU-bound operation.

The sync and subscription timeouts are different. Both involve k-many _network_ operations. They're like the opposite, basically.

Should the storage observer timeout be more like ~2 seconds?

Reply:
^ Ah good call out, though. We should find out whether or not this triggers loading of the queue from SQL. A disk-bound timeout will be different than a CPU-bound timeout.